### PR TITLE
Add GraphTools to Graph Notebook and update other notebooks

### PR DESCRIPTION
**Pull request by cndolo**
This PR adds some examples for the `GraphTools` class in the Graph notebook.
While at it, I updated the notebooks that were using the newly deprecated functions.

### CODE REVIEW

File: notebooks/GraphNotebook.ipynb [hunk 1]
```diff
@@ -287,6 +289,114 @@
...
+   "metadata": {},
+   "source": [
+    "## GraphTools"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Some useful methods that give information on graphs or change graphs are not implemented in the graph class itself, but rather in the `GraphTools` class. The following section demonstrates some of the GraphTools functionalities."
```
Comment (angriman) [inline notebooks/GraphNotebook.ipynb]:
```suggestion
    "Some useful methods that give information on graphs or change graphs are implemented in `networkit.graphtools`. The following section demonstrates some of the GraphTools functionalities."
```

File: notebooks/GraphNotebook.ipynb [hunk 2]
```diff
@@ -287,6 +289,114 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The `toWeighted(G)` function transforms an unweighted graph to a weighted graph, and returns the weighted copy of the input graph."
```
Comment (angriman) [inline notebooks/GraphNotebook.ipynb]:
```suggestion
    "The function `toWeighted(G)` takes an unweighted graph `G` as input, and returns a weighted copy of `G`. All the edge weights are set to a default weight of 1.0."
```

File: notebooks/GraphNotebook.ipynb [hunk 3]
```diff
@@ -287,6 +289,114 @@
...
+    "weightedG = nk.graphtools.toWeighted(G)\n",
+    "assert(weightedG.numberOfNodes() == G.numberOfNodes())\n",
+    "assert(weightedG.numberOfEdges() == G.numberOfEdges())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The to `toUnweighted(G)` function then does the inverse of the above."
```
Comment (angriman) [inline notebooks/GraphNotebook.ipynb]:
```suggestion
    "The function `toUnweighted(G)` does the inverse of the one above: it takes a weighted graph `G` as input, and returns an unweighted copy of `G` as output."
```

File: notebooks/GraphNotebook.ipynb [hunk 4]
```diff
@@ -287,6 +289,114 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "One can query the `randomNeighbor(G, u)` method for a random neighbor of node `u` by passing a graph and the node `u`."
```
Comment (angriman) [inline notebooks/GraphNotebook.ipynb]:
```suggestion
    "The function`randomNeighbor(G, u)` returns a random neighbor of node `u` in the graph `G`."
```

File: notebooks/GraphNotebook.ipynb [hunk 5]
```diff
@@ -287,6 +289,114 @@
...
+   "source": [
+    "random = nk.graphtools.randomNeighbor(G, 0)\n",
+    "print(random)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Sometimes it makes sense to compact the graph, e.g after deleting nodes. The method `getCompactedGraph(G, nodeIdMap)` does just that by designating continous node ids. `nodeIdMap` is a map providing information about the node ids.\n",
```
Comment (angriman) [inline notebooks/GraphNotebook.ipynb]:
```suggestion
    "Sometimes it makes sense to compact the graph, e.g., after deleting nodes. The method `getCompactedGraph(G, nodeIdMap)` does just that by designating continuous node ids. `nodeIdMap` maps each node id of graph `G` to their new ids.\n",
```

File: notebooks/GraphNotebook.ipynb [hunk 6]
```diff
@@ -287,6 +289,114 @@
...
+    "\n",
+    "First, we will delete a node from `G`."
```
Comment (angriman) [inline notebooks/GraphNotebook.ipynb]:
```suggestion
    "First, we delete a node from `G`."
```

File: notebooks/GraphNotebook.ipynb [hunk 7]
```diff
@@ -287,6 +289,114 @@
...
+   "outputs": [],
+   "source": [
+    "G.removeNode(2)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Then compute a map of node ids."
```
Comment (angriman) [inline notebooks/GraphNotebook.ipynb]:
```suggestion
    "Then, we use `getContinuousNodeIds(G)` to get a map from the original nodes ids of G, to their new ids."
```

File: notebooks/GraphNotebook.ipynb [hunk 8]
```diff
@@ -287,6 +289,114 @@
...
+   "outputs": [],
+   "source": [
+    "nodeIdMap = nk.graphtools.getContinuousNodeIds(G)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can now get the compacted graph with the same structure as the original graph."
```
Comment (angriman) [inline notebooks/GraphNotebook.ipynb]:
```suggestion
    "Finally, we get a new graph with compacted node ids."
```

**Comment by cndolo** [comment]
Thank you for the review; I have made the changes.

**Pull request merged**